### PR TITLE
task(SDK-3814) - Moves persistence after inapp evaluation

### PR DESCRIPTION
**Pull request by Anush-Shand**
- Delays the persistence to after inapp evaluation
- Constructs a map of userAtttibutes to be used for evaluation

### CODE REVIEW

File: clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+import com.clevertap.android.sdk.ProfileValueHandler.NumberValueType.FLOAT_NUMBER
+import com.clevertap.android.sdk.ProfileValueHandler.NumberValueType.INT_NUMBER
+import com.clevertap.android.sdk.utils.CTJsonConverter
+import com.clevertap.android.sdk.validation.ValidationResult
+import com.clevertap.android.sdk.validation.ValidationResultFactory
+import com.clevertap.android.sdk.validation.ValidationResultStack
+import com.clevertap.android.sdk.validation.Validator
+import org.json.JSONArray
+
+class ProfileValueHandler(
```
Comment (CTLalit) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt:13]: does this contain moved code from analytics manager and and java -> kt, is there any change in logic?

File: clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+) {
+
+    internal enum class NumberValueType {
+        INT_NUMBER,
+        FLOAT_NUMBER,
+        DOUBLE_NUMBER
+    }
+
+    private var numberValueType: NumberValueType? = null
+    fun handleIncrementDecrementValues(value: Number, command: String, existingValue: Number?): Number? {
```
Comment (CTLalit) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt:25]: can we handle nullables here in better way?

File: clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            numberValueType = INT_NUMBER
+        } else if (value == value.toDouble()) {
+            numberValueType = DOUBLE_NUMBER
+        } else if (value == value.toFloat()) {
+            numberValueType = FLOAT_NUMBER
+        }
+        return numberValueType
+    }
+
+    fun computeMultiValues(key: String, values: JSONArray?, command: String, existingValues: Any?): JSONArray? {
```
Comment (CTLalit) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt:88]: test cases can be covered

File: clevertap-core/src/main/java/com/clevertap/android/sdk/events/EventMediator.java [hunk 4]
```diff
@@ -83,6 +98,10 @@ public boolean isEvent(JSONObject event) {
         return event.has(Constants.KEY_EVT_NAME);
     }
 
+    public boolean isProfileEvent(JSONObject event) {
+        return event.has(Constants.KEY_EVT_NAME);
+    }
```
Comment (CTLalit) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/events/EventMediator.java:103]: check if this is needed

File: clevertap-core/src/main/java/com/clevertap/android/sdk/Constants.java [hunk 5]
```diff
@@ -391,6 +397,8 @@ public interface Constants {
     HashSet<String> NONE_CRYPT_KEYS = new HashSet<>(Arrays.asList(KEY_ENCRYPTION_MIGRATION));
     HashSet<String> piiDBKeys = new HashSet<>(Arrays.asList(KEY_ENCRYPTION_NAME, KEY_ENCRYPTION_EMAIL, KEY_ENCRYPTION_IDENTITY, KEY_ENCRYPTION_PHONE));
 
+    HashSet<String> keysToSkipForUserAttributesEvaluation = new HashSet<>(Arrays.asList("cc", "tz", "Carrier"));
```
Comment (CTLalit) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/Constants.java:400]: check if there are more computed sdk attributes which we need to ignore.

File: clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt (hunk 1 above)
Comment (Anush-Shand) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt:13]: Logic is the same. Some top-level functions have been added
Other functions have been moved from AnalyticsManager.java

File: clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt (hunk 3 above)
Comment (Anush-Shand) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt:88]: Will add test cases for the additions

File: clevertap-core/src/main/java/com/clevertap/android/sdk/Constants.java (hunk 5 above)
Comment (Anush-Shand) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/Constants.java:400]: Currently only 3 which are added from `pushBasicProfile` in `EventQueueManager`

File: clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt (hunk 2 above)
Comment (Anush-Shand) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt:25]: Will check if refactoring is needed for the older logic

**Pull request merged**